crates/pevm/src/storage/rpc.rs

```diff
@@ -51,36 +48,32 @@ impl<N: Network> RpcStorage<N> {
             cache_accounts: Mutex::default(),
             cache_bytecodes: Mutex::default(),
             cache_block_hashes: Mutex::default(),
-            // TODO: Better error handling.
-            runtime: Runtime::new().unwrap(),
         }
     }
 
     /// Send a request and retry many times if needed.
     /// This util is made to avoid error 429 Too Many Requests
-    /// <https://en.wikipedia.org/wiki/Exponential_backoff>
-    fn fetch<T, E, R: IntoFuture<Output = Result<T, E>>>(
+    /// https://en.wikipedia.org/wiki/Exponential_backoff
+    async fn fetch<T, E, R: IntoFuture<Output = Result<T, E>>>(
         &self,
         request: impl Fn() -> R,
     ) -> Result<T, E> {
         const RETRY_LIMIT: usize = 8;
         const INITIAL_DELAY_MILLIS: u64 = 125;
 
-        self.runtime.block_on(async {
-            let mut lives = RETRY_LIMIT;
-            let mut delay = Duration::from_millis(INITIAL_DELAY_MILLIS);
-
-            loop {
-                let result = request().await;
-                if lives > 0 && result.is_err() {
-                    tokio::time::sleep(delay).await;
-                    lives -= 1;
-                    delay *= 2;
-                } else {
-                    return result;
-                }
+        let mut lives = RETRY_LIMIT;
+        let mut delay = Duration::from_millis(INITIAL_DELAY_MILLIS);
+
+        loop {
+            let result = request().await;
+            if lives > 0 && result.is_err() {
+                sleep(delay).await;
```

Comment:
Let's quantify common names like `sleep` and `spawn` as the difference between `std` and `tokio` (or other libraries) can be confusing.